src/core/meshcop/dtls.hpp

```diff
@@ -247,6 +356,26 @@ class Dtls : public InstanceLocator
     uint8_t mPsk[kPskMaxLength];
     uint8_t mPskLength;
 
+#if OPENTHREAD_ENABLE_APPLICATION_COAP_SECURE
+
+    mbedtls_x509_crt   mCaCert;
+    mbedtls_pk_context mPrivateKey;
+    const uint8_t *    mPk;
+    uint32_t           mPkLength;
+    const uint8_t *    mX509Cert;
+    uint32_t           mX509CertLength;
+
+    uint8_t  mPreSharedKey[kPreSharedKeyMaxLength];
```

Comment:
Do we need to maintain a copy here?  Or can we just rely on the pointer provided by the user?

Reply:
No, it is not necessary. Now we use the users pointer.
Done.